### PR TITLE
Connection banner: remove B test since it's no longer used

### PR DESCRIPTION
**Pull request by eliorivero**
Until 5.3, there was an A/B test running. It's been removed now during the Shortcopy project where we focused Jetpack on Security and Performance.

#### Changes proposed in this Pull Request:

* Remove all code and logic specific for the A/B test since we only have one now.

#### Testing instructions:

* Test the connect banner in Plugins or Dashboard screens in WP Admin, everything should work normally.

#### Proposed changelog entry for your changes:

* Not needed. Part of Shortcopy.

### CODE REVIEW

**Comment by kraftbj** [comment]
@eliorivero Can you update the unit test?

```
There was 1 failure:
1) WP_Test_Jetpack_Connection_Banner::test_connection_banner_cleans_up_on_upgrade
Failed asserting that 2 is false.
/tmp/wordpress-master/src/wp-content/plugins/jetpack/tests/php/test_class.jetpack-connection-banner.php:13```

File: class.jetpack-connection-banner.php [hunk 1]
```diff
@@ -22,50 +22,6 @@ static function init() {
 	 */
 	private function __construct() {
 		add_action( 'current_screen', array( $this, 'maybe_initialize_hooks' ) );
-		add_action( 'updating_jetpack_version', array( $this, 'cleanup_on_upgrade' ), 10, 2 );
-	}
-
-	function cleanup_on_upgrade( $new_version = null, $old_version = null ) {
-		if ( version_compare( $old_version, '4.4', '>=' ) && version_compare( $old_version, '5.3', '<' ) ) {
-			delete_option( 'jetpack_connection_banner_ab' );
-		}
```
Comment (enejb) [inline class.jetpack-connection-banner.php]: Would there be any cases where a user might have this option saved somewhere. We might want to add it to the list of options to be deleted when the user deletes jetpack.

Comment (eliorivero) [reply class.jetpack-connection-banner.php]: Option is deleted, checked 

https://github.com/Automattic/jetpack/blob/master/class.jetpack-options.php#L535

and

https://github.com/Automattic/jetpack/blob/master/class.jetpack-options.php#L576

Thanks for bringing this up.

**Comment by eliorivero** [comment]
Test updated. I removed the file since it was only used for that method.

**Review by kraftbj — APPROVED** [review]

**Comment by jeffgolenski** [comment]
Thanks Elio!

**Pull request merged**